### PR TITLE
refactor: Port settings menu

### PR DESCRIPTION
**Pull request by kartikCypherock**
This pull request implements settings menu on the X1 vault device

Implements https://app.clickup.com/t/9002019994/PRF-4854

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4854 Port settings menu](https://app.clickup.com/t/9002019994/PRF-4854)

File: src/constant_texts.h [hunk 1]
```diff
@@ -59,10 +61,20 @@ extern const char *ui_text_options_new_wallet[];
 
 // Settings menu text
 #ifdef DEV_BUILD
-#define NUMBER_OF_OPTIONS_ADVANCED_OPTIONS 10
+#define NUMBER_OF_OPTIONS_SETTINGS 10
+#define NUMBER_OF_OPTIONS_ADVANCED_OPTIONS NUMBER_OF_OPTIONS_SETTINGS
```
Comment (ujjwal-cyph) [inline src/constant_texts.h:65]:
```suggestion
// TODO: Update after refactor - remove the following MACRO
#define NUMBER_OF_OPTIONS_ADVANCED_OPTIONS NUMBER_OF_OPTIONS_SETTINGS
```

File: src/constant_texts.h [hunk 2]
```diff
@@ -59,10 +61,20 @@ extern const char *ui_text_options_new_wallet[];
...
 #else
-#define NUMBER_OF_OPTIONS_ADVANCED_OPTIONS 9
+#define NUMBER_OF_OPTIONS_SETTINGS 9
+#define NUMBER_OF_OPTIONS_ADVANCED_OPTIONS NUMBER_OF_OPTIONS_SETTINGS
```
Comment (ujjwal-cyph) [inline src/constant_texts.h:68]:
```suggestion
// TODO: Update after refactor - remove the following MACRO
#define NUMBER_OF_OPTIONS_ADVANCED_OPTIONS NUMBER_OF_OPTIONS_SETTINGS
```

File: src/menu/settings_menu.c [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+ * @param p0_evt The p0 event object which triggered the callback
+ * @param data_ptr Currently unused pointer set by the engine
+ */
+static void ignore_p0_handler(engine_ctx_t *ctx,
+                              p0_evt_t p0_evt,
+                              const void *data_ptr);
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+const flow_step_t settings_menu_step = {
```
Comment (ujjwal-cyph) [inline src/menu/settings_menu.c:138]: make the variable `static`

Comment (ujjwal-cyph) [reply src/menu/settings_menu.c:138]:
```suggestion
static const flow_step_t settings_menu_step = {
```

File: src/menu/settings_menu.c [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        // TODO: Handle all cases
+        break;
+      }
+    }
+  } else {
+    // UI_EVENT_LIST_REJECTION handled below already
+  }
+
+  /* Return to the previous menu irrespective if UI_EVENT_REJECTION was
+   * detected, or a create wallet flow was executed */
```
Comment (ujjwal-cyph) [inline src/menu/settings_menu.c:191]:
```suggestion
   * detected, or any options was executed */
```

**Review by ujjwal-cyph — APPROVED** [review]

**Pull request merged**